commcare_connect/opportunity/api/views.py

```diff
@@ -43,3 +52,27 @@ class UserVisitViewSet(viewsets.GenericViewSet, viewsets.mixins.ListModelMixin):
 
     def get_queryset(self):
         return UserVisit.objects.filter(opportunity=self.kwargs.get("opportunity_id"), user=self.request.user)
+
+
+class ClaimOpportunityView(APIView):
+    authentication_classes = [OAuth2Authentication]
+    permission_classes = [TokenHasReadWriteScope]
+
+    def post(self, *args, **kwargs):
+        opportunity_access = get_object_or_404(OpportunityAccess, user=self.request.user, opportunity=kwargs.get("pk"))
+        opportunity = opportunity_access.opportunity
+        if opportunity.learn_app.cc_domain != opportunity.deliver_app.cc_domain:
+            create_hq_user(self.request.user, opportunity.deliver_app.cc_domain, opportunity.api_key)
+
+        claim, created = OpportunityClaim.objects.get_or_create(
+            opportunity_access=opportunity_access,
+            defaults={
+                "max_payments": opportunity.daily_max_visits_per_user,
+                "end_date": opportunity.end_date,
+            },
+        )
+
+        if not created:
+            return Response(status=400, data="Opportunity is already claimed")
```

Comment:
Maybe we just return a 200 here instead of an error since re-claiming isn't really an issue?